plugins/status.js

```diff
@@ -0,0 +1,241 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const async = require('async');
+
+exports.register = function () {
+    this.outbound = require('./../outbound');
+    this.queue_dir = require('./../outbound/queue').queue_dir;
+}
+
+exports.hook_capabilities = function (next, connection) {
+
+    if (connection.remote.is_local) {
+        connection.capabilities.push('STATUS');
+    }
+
+    next();
+}
+
+exports.hook_unrecognized_command = function (next, connection, params) {
+    if (params[0] !== 'STATUS') {
+        next();
+    }
+    else if (!connection.remote.is_local) {
+        return next(DENY, "STATUS not allowed remotely");
+    }
+    else {
+        return this.run(params[1], function (err, result) {
+
+            if (err) return next(DENY, err.message);
+            return connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
+        })
+    }
+}
```

Comment:
Within this function you have inconsistent use of `return`. Return stops the execution of a function and returns a value. We generally use `return` for two purposes: flow control and/or returning something (a value, a promise, etc.) to the calling function.

Since nothing is checking the return values, it's obvious that you're using `return` for flow control. But you're also using the `if` blocks for flow control. Your first `if` condition doesn't have a `return`, but the 2nd and 3rd do. You should choose one or the other for flow control. The result would look like one of:

## return for flow control

```js
exports.hook_unrecognized_command = function (next, connection, params) {
    if (params[0] !== 'STATUS')      return next();
    if (!connection.remote.is_local) return next(DENY, "STATUS not allowed remotely");

    this.run(params[1], function (err, result) {

        if (err) return next(DENY, err.message);
        connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
    })
}
```

## conditional blocks for flow control

```js
exports.hook_unrecognized_command = function (next, connection, params) {
    if (params[0] !== 'STATUS') {
        next();
    }
    else if (!connection.remote.is_local) {
        next(DENY, "STATUS not allowed remotely");
    }
    else {
        this.run(params[1], function (err, result) {

            if (err) return next(DENY, err.message);
            connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
        })
    }
}
```

Choose your favorite. I prefer the former because I find it easier to read, nicer to extend (over time, the stuff inside the if test(s) tends to grow, making them hairy), and since JS often forces indents on us, I like to avoid them when I can.

Reply:
Yep, these two ifs are just validation before actual work. I don't know why I mixed this, I like former way too...